File: ASP.NETCoreWebApplication1/Services/IGameService.cs
using Microsoft.AspNetCore.Mvc;
using Microsoft.AspNetCore.Mvc.Infrastructure;
using Microsoft.AspNetCore.SignalR;

namespace ASP.NETCoreWebApplication1.Services;

public interface IGameService<T>
{
    public GameBase<IGameService<T>> Start();

    public Task<GameBase<IGameService<T>>> StartAsync();

    public GameBase<IGameService<T>> Suspend(Double time);

    public Task<GameBase<IGameService<T>>> SuspendAsync(Double time);

    public GameBase<IGameService<T>> Finish();

    public Task<GameBase<IGameService<T>>> FinishAsync();

}

File: TexasHoldem.Models/Services/IGameService.vb
Imports TexasHoldem.Models.Services

Public Interface IGameService(Of T)
    Public Function Start() As GameBase(Of IGameService(Of T))
    Public Function StartAsync() As Task(Of GameBase(Of IGameService(Of T)))
    Public Function Suspend(ByVal time As Double) As GameBase(Of IGameService(Of T))
    Public Function SuspendAsync(ByVal time As Double) As Task(Of GameBase(Of IGameService(Of T)))
    Public Function Finish() As GameBase(Of IGameService(Of T))
    Public Function FinishAsync() As Task(Of GameBase(Of IGameService(Of T)))
    
End Interface